index/scorch/segment/zap/dict.go

```diff
@@ -122,23 +122,33 @@ func (d *Dictionary) PrefixIterator(prefix string) segment.DictionaryIterator {
 		d: d,
 	}
 
+	kBeg := []byte(prefix)
+	kEnd := incrementBytes(kBeg)
+
 	if d.fst != nil {
-		r, err := regexp.New(prefix + ".*")
+		itr, err := d.fst.Iterator(kBeg, kEnd)
 		if err == nil {
-			itr, err := d.fst.Search(r, nil, nil)
-			if err == nil {
-				rv.itr = itr
-			} else if err != nil && err != vellum.ErrIteratorDone {
-				rv.err = err
-			}
-		} else {
+			rv.itr = itr
+		} else if err != nil && err != vellum.ErrIteratorDone {
 			rv.err = err
 		}
 	}
 
 	return rv
 }
 
+func incrementBytes(in []byte) []byte {
+	rv := make([]byte, len(in))
+	copy(rv, in)
```

Comment:
just a doubt, copy vs append, any preferences?

Reply:
yah, that was the old, working code straight from upside-down/moss, so figured just take it as-is for now.  Originally from...

https://github.com/blevesearch/bleve/blob/master/index/store/moss/reader.go#L87

Reply:
and, indeed, i would suspect append() might be faster, in that it would push more of the operation into the hands of the golang runtime, which might be able to optimize better

but, since this only happens once per query, also less worried about it